[HttpFoundation] Rename Request::getContentType to getContentTypeFormat

MarkPedron opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | yes
| Tickets       | Fix #39750
| License       | MIT
| Doc PR        | TODO

MarkPedron committed:
Rename getContentType to getContentTypeFormat

Resolves issue #39750.

The method getContentType was confusing.
This method does not return a mime type, but a mapped type name derived
from the mime type in the CONTENT_TYPE header.

wouterj commented on src/Symfony/Component/HttpFoundation/Request.php:
```diff
@@ -1324,8 +1324,19 @@ public function setRequestFormat(?string $format)
+    {
+        trigger_deprecation('symfony/http-foundation', '6.1', 'The method %s is deprecated, use "getContentTypeFormat" instead.', __METHOD__);
+
+        return $this->getFormat($this->headers->get('CONTENT_TYPE', ''));
```
```suggestion
        return $this->getContentTypeFormat();
```

MarkPedron replied:
Fixed (ammended commit)

wouterj commented on src/Symfony/Component/HttpFoundation/Request.php:
```diff
@@ -1324,8 +1324,19 @@ public function setRequestFormat(?string $format)
      */
     public function getContentType(): ?string
+    {
+        trigger_deprecation('symfony/http-foundation', '6.1', 'The method %s is deprecated, use "getContentTypeFormat" instead.', __METHOD__);
```
```suggestion
        trigger_deprecation('symfony/http-foundation', '6.1', 'The method "%s" is deprecated, use "getContentTypeFormat" instead.', __METHOD__);
```

MarkPedron replied:
Fixed, including necessary test changes

wouterj reviewed (commented):
Looks good from my point of view :+1:

MarkPedron commented:
@wouterj Thank you for your kind feedback.

Sorry for all the updates, I completely forgot to add tests and the necessary deprecation infos.

Should be ready now. The CI failure is very mysterious to me.

wouterj commented:
No problem! For a first contribution, this is in perfect shape (especially considering changes like this need BC and deprecations and stuff).

MarkPedron commented:
I made an attempt to make appveyor happy (I think i understood the subsequent deprecation emitted in security/http).
The errors in the php8.0/php8.1 job also seem understandable (the return type patch does not apply, because the code changed). However, they seem unrelated to this PR/issue.

stof commented on src/Symfony/Component/HttpFoundation/Tests/RequestTest.php:
```diff
@@ -78,14 +81,27 @@ public function testIsNoCache()
+        $request = new Request();
+        $contentType = $request->getContentTypeFormat();
+
+        $this->assertNull($contentType);
```
Would be great to add more tests covering cases where that method returns a non-null value. Otherwise, we don't really have tests for the feature.

MarkPedron replied:
Thank you for the suggestion. I added tests to cover the mime types `application/json` and `text/html`

chalasr commented on src/Symfony/Component/HttpFoundation/Request.php:
```diff
@@ -1324,8 +1324,20 @@ public function setRequestFormat(?string $format)
+    /**
+     * Gets the format associated with the request.
+     */
+    public function getContentTypeFormat(): ?string
```
I agree with what this aims to fix, but the difference between `getRequestFormat()` and `getFormat()` was already not obvious before this PR. Adding `getContentTypeFormat` makes the API even more confusing. We should probably make the difference clear with PHPDoc at least

ogizanagi commented on src/Symfony/Component/HttpFoundation/Request.php:
```diff
@@ -1324,8 +1324,20 @@ public function setRequestFormat(?string $format)
+    }
+
+    /**
+     * Gets the format associated with the request.
```
```suggestion
     * Gets the usual name of the format associated with the request Content-Type mime type.
     *
     * @see Request::$formats
```

?

nicolas-grekas commented:
Friendly ping @MarkPedron, up to finish this PR? See pending comments + rebase needed.

MarkPedron commented:
@nicolas-grekas Thanks for the reminder. I added non-null tests as suggested, accepted the suggestion expanding the comment, and rebased. If there is anything else to be done, please let me know. I am hesitating to expand the comments further, as to me the meaning of getContentTypeFormat seems quite clear, and confusion arises from the methods setRequestFormat/getRequestFormat, which (to my understanding) refer to the negotiated format of the _Response_.

ihmels commented on UPGRADE-6.1.md:
```diff
@@ -27,6 +27,11 @@ FrameworkBundle
+HttpFoundation
+--------------
+
+ * Deprecate `Request::getContentType()`, use `Request::getContentTypeFormat()` instead`
```
There is a trailing backtick that needs to be removed.

ihmels replied:
And this point needs to be moved to UPGRADE-6.2.md.

ihmels commented on src/Symfony/Component/HttpFoundation/Request.php:
```diff
@@ -1326,8 +1326,22 @@ public function setRequestFormat(?string $format)
+    }
+
+    /**
+     * Gets the usual name of the format associated with the request Content-Type mime type.
```
```suggestion
     * Gets the usual name of the format associated with the request's media type (provided in the Content-Type header).
```

fabpot commented:
Honestly, I don't think this is worth it. Maybe adding copious docblocks to explain what the method does (and a few others like explained in this [comment](https://github.com/symfony/symfony/issues/39750#issuecomment-757207375) would be more efficient).

fabpot commented:
Thank you @MarkPedron.
